src/shared/regions/defaultRegionProvider.ts

```diff
@@ -4,98 +4,59 @@
  */
 
 import * as vscode from 'vscode'
-import { endpointsFileUrl } from '../constants'
-import { ext } from '../extensionGlobals'
-import { getLogger, Logger } from '../logger'
-import { CompositeResourceFetcher } from '../resourcefetcher/compositeResourceFetcher'
-import { FileResourceFetcher } from '../resourcefetcher/fileResourceFetcher'
-import { HttpResourceFetcher } from '../resourcefetcher/httpResourceFetcher'
-import { ResourceFetcher } from '../resourcefetcher/resourcefetcher'
+import { Endpoints, Region } from './endpoints'
+import { EndpointsProvider } from './endpointsProvider'
 import { RegionInfo } from './regionInfo'
 import { RegionProvider } from './regionProvider'
 
-export interface RawRegion {
-    description: string
-}
+export class DefaultRegionProvider implements RegionProvider {
+    private readonly onRegionProviderUpdatedEmitter: vscode.EventEmitter<void> = new vscode.EventEmitter()
```

Comment:
nit: `Emitter` isn't needed

Reply:
onRegionProviderUpdatedEmitter is the emitter that produces events
onRegionProviderUpdated is the event that callers can register with